src/player.cpp

```diff
@@ -5661,16 +5661,18 @@ void player::try_to_sleep( const time_duration &dur )
     bool watersleep = false;
     if( has_trait( trait_CHLOROMORPH ) ) {
         plantsleep = true;
-        if( ( ter_at_pos == t_dirt || ter_at_pos == t_pit ||
-              ter_at_pos == t_dirtmound || ter_at_pos == t_pit_shallow ||
-              ter_at_pos == t_grass ) && !vp &&
-            furn_at_pos == f_null ) {
-            add_msg_if_player( m_good, _( "You relax as your roots embrace the soil." ) );
-        } else if( vp ) {
+        // Down to bottom: if in vehicle - can't sleep
+        if( vp ) {
             add_msg_if_player( m_bad, _( "It's impossible to sleep in this wheeled pot!" ) );
-        } else if( furn_at_pos != f_null ) {
+        // Else if on wrong furniture - can't sleep
+        } else if( furn_at_pos != f_null && !( g->m.has_flag_furn( "ALLOW_ROOT", pos() ) ||  g->m.has_flag_furn( "ALLOW_ROOT_WEAK", pos() ) ) ) {
```

Comment:
I'm not sure I understand the logic here - if the furniture is not null, and doesn't have the `ALLOW_ROOT` flag, or if it does have the `ALLOW_ROOT_WEAK`, you can't use your roots to sleep on it.
But then the check after it allows you to sleep if the furniture has the `ALLOW_ROOT` or `ALLOW_ROOT_WEAK` flag.

So, it won't be possible to use your roots on a furniture with `ALLOW_ROOT_WEAK`.

Reply:
I was under impression that statement was:
if furniture NOT null AND NOT (furniture with allow_root OR furniture with allow_root_weak)
So, if furniture has at least on of the flags this would return true and we should skip this case.
ALLOW_ROOT_WEAK should allow to sleep it just represents that it is not as comfortable as it could be.

Reply:
I also missed the parenthesis, sorry.